### DISCUSSION TITLE
本地文件 xlsx pdf docx 等无法预览

### DISCUSSION THREAD

**Discussion by zfxkj** [AlistGo/alist#594] [Q&A] [answered]
我用docker搭建的，上传文件到本地后其中音频，视频，图片等均可正常预览，office文件不能正常预览
显示：
An error occurred
We're sorry, but for some reason we can't open this for you.
我在github没有找到解决方案
希望知道的小伙伴可以告诉我解决方案，谢谢
![image](https://user-images.githubusercontent.com/31238863/155084739-0cfcd55f-2364-4558-a5ab-bd8a912d33e9.png)

**Comment by xhofe** [comment, marked as answer]
office 在线预览限制：

- 外网可访问
- 文档访问地址不能直接使用 ip，需要通过域名访问，并且端口必须是 80 / 443 端口
- 文档的格式（必须为以下格式之一）：
Word：docx、docm、dotm、dotx
Excel：xlsx、xlsb、xls、xlsm
PowerPoint：pptx、ppsx、ppt、pps、pptm、potm、ppam、potx、ppsm
- 文档的大小：Word 和 PowerPoint 文档必须小于 10 兆字节;Excel 必须小于五兆字节

**Reply by zfxkj** [reply]
OK，解决了

**Reply by StudentWeis** [reply]
IP访问和端口限制这个可以改进嘛？

**Reply by wu2305** [reply]
> IP访问和端口限制这个可以改进嘛？

这个是微软的预览服务的限制，如果要更改请自己搭建onlyoffice，或者自己想办法集成其他文件预览工具

**Reply by suyinLoong** [reply]
我这里其它条件都符合，但是我是使用 docker-compose 安装在服务器上的，端口默认是 5244，现在预览 office 文件报错：
```
File not found
The URL of the original file is not valid or the document is not publicly accessible.
Verify the URL is correct, then contact the document owner.
```
这个要怎么办？

**Reply by Simonzhang8** [reply]
> 我这里其它条件都符合，但是我是使用 docker-compose 安装在服务器上的，端口默认是 5244，现在预览 office 文件报错：
> 
> ```
> File not found
> The URL of the original file is not valid or the document is not publicly accessible.
> Verify the URL is correct, then contact the document owner.
> ```
> 
> 这个要怎么办？

给个具体的地址

**Reply by suyinLoong** [reply]
> > 我这里其它条件都符合，但是我是使用 docker-compose 安装在服务器上的，端口默认是 5244，现在预览 office 文件报错：
> > ```
> > File not found
> > The URL of the original file is not valid or the document is not publicly accessible.
> > Verify the URL is correct, then contact the document owner.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 这个要怎么办？
> 
> 给个具体的地址

已解决，谢谢！

**Comment by dadigang** [comment]
o

**Comment by liugensite** [comment]
office的在线预览接口是不是不支持IPV6

**Comment by wu2305** [comment]
是的，不支持纯v6-------- Original message --------From: Nucleus ***@***.***>Date: Tue, Nov 22, 2022, 10:10To: alist-org/alist ***@***.***>Cc: wu2305 ***@***.***>, Comment ***@***.***>Subject: Re: [alist-org/alist] 本地文件 xlsx pdf docx 等无法预览 (Discussion #594)

office的在线预览接口是不是不支持IPV6

**Comment by liugensite** [comment]
好的，感谢！

wu2305 ***@***.***> 于2022年11月22日周二 12:06写道：

**Comment by qinguoluo** [comment]
我的alist部署在自己的电脑上，不能预览office 文档，能解决吗？

**Reply by qinguoluo** [reply]
文档访问地址不能直接使用 ip，需要通过域名访问，并且端口必须是 80 / 443 端口

不懂，请大佬指点一下，具体该怎么做？

**Reply by Simonzhang8** [reply]
最重要的是相应链接公网能够访问，这方面得自己解决，能公网访问了其他的就好办

**Comment by wujiaren** [comment]
有人遇到过打开10M以上的文件会乱码的吗，通过raidrive挂载到本地然后打开，挂载的是自研的网盘

**Comment by kevinsheep** [comment]
我尝试将AList部署到80或443端口了，服务也是通过域名访问，并且文件权限设置为guest免登录访问。使用google预览可以的，但要翻墙；但microsoft一直不行，提示文件无法访问：

> File not found
> The URL of the original file is not valid or the document is not publicly accessible.
> Verify the URL is correct, then contact the document owner.

**Reply by kevinsheep** [reply]
已自己解决。
直接使用 [https://view.officeapps.live.com/op/view.aspx?src=xxx](https://view.officeapps.live.com/op/view.aspx?src=xxx) 打开文件预览，发现一切正常，再仔细查看，发现在 AList 点开的预览页，地址确实不太正常。
将 `$e_url` 改成 `$durl` 后，预览功能暂时正常使用。特意使用带中文的文件名也能正常预览。

**Reply by Luci8109** [reply]
折腾了一圈，还是你这个最有效，完美

**Reply by lizhiliao** [reply]
> 已自己解决。 直接使用 https://view.officeapps.live.com/op/view.aspx?src=xxx 打开文件预览，发现一切正常，再仔细查看，发现在 AList 点开的预览页，地址确实不太正常。 将 `$e_url` 改成 `$durl` 后，预览功能暂时正常使用。特意使用带中文的文件名也能正常预览。

使用这个方法可以在线预览office了，不过pdf还是不行

**Comment by cui8** [comment]
> 已自己解决。 直接使用 https://view.officeapps.live.com/op/view.aspx?src=xxx 打开文件预览，发现一切正常，再仔细查看，发现在 AList 点开的预览页，地址确实不太正常。 将 `$e_url` 改成 `$durl` 后，预览功能暂时正常使用。特意使用带中文的文件名也能正常预览。

大佬，能分享下【Iframe 预览】配置吗？

**Comment by kevinsheep** [comment]
> > 已自己解决。 直接使用 https://view.officeapps.live.com/op/view.aspx?src=xxx 打开文件预览，发现一切正常，再仔细查看，发现在 AList 点开的预览页，地址确实不太正常。 将 `$e_url` 改成 `$durl` 后，预览功能暂时正常使用。特意使用带中文的文件名也能正常预览。
> 
> 大佬，能分享下【Iframe 预览】配置吗？

只将 $e_url 改成 $durl ，其他没什么改动

**Reply by jarodvip** [reply]
没有什么改变，还是无法使用
